### DISCUSSION TITLE
Requester Interrupts not working

### DISCUSSION THREAD

**Discussion by infinite-loop0** [Opinionated-Blueprints/10-Books-Full-of-Rails#114] [Q&A]
So I have an issue with the latest version 2.21.2.

TLDR - entire system is working except the requester process. The trains fill correctly at a provider station but then sit there indefinitely in a hanging state, irrespective of the requester station status. They appear to be guided by the status of a signal 'M', which I cannot understand where this is set or what sets it. There may be something more to this - please advise!

I have used the blueprints to make trains of 2-4-0 structure. This is all for solids at this point. I have set up a range of requester and provider stations using the blueprints again. I also have a few depot stops too. Stackers/max number has been set [per the guide](https://github.com/Opinionated-Blueprints/10-Books-Full-of-Rails/wiki/Provider-and-Requester-stations).

The stations for fuelling, fuel-loading, depot and provider all work as expected - trains fuel, the refuel train works completely, and the trains are dispatched in appropriate numbers to the stackers of the provider stations from either requesters when manually called, or the depot entirely autonomously.

However, once the train at a provider is full, nothing happens. The train just sits there and the interrupt is never actioned.

If you manually click the interrupt '<Group><TYPE>Requester' to run it to the requester station, the train then route plans correctly, to the right station and unloads, followed by an appropriate return to either a depot or a provider station with a stacker - which suggests the entire system is working except the call from the provider to the requester.

The requester stations never work automatically - even when entirely empty. It appears the behaviour is all mediated through signal 'M', which is not documented in the Wiki, and I cannot find it in any of the blueprints to explain how m=-1 or not -1 is set. I have tried removing and replacing all blueprints, and all trains without success.

**Comment by oliverrahner** [comment]
I had the same (or a very similar) issue.

**Cause**
TL;DR: You trains' schedule is broken.

What I found is that I renamed some stations to clean up some mess I made before.
While renaming, Factorio automatically keeps the references in train's schedules up-to-date, by renaming those as well.
Depending on what exactly you changed, it might have caused collisions. In my case, the provider interrupt routed the trains to a _requester_ station instead, so the opposite of what you are seing.

You tried the right thing by replacing the trains with new ones, **but** the schedules for a group of trains is stored globally, not for each train. So if the schedule is broken and you replace the trains one by one, it will _stay_ broken.

**Fix**
TL;DR: You need to fix the schedule :)

**Option 1:** Remove **all trains of the same group**, _then_ recreate them
**Option 2:** Introduce new groups, so you avoid touching any existing schedule.
**Option 3:** Fix the current schedules manually. You could create a new train from a blueprint for a new group and then visually copy-paste the settings.

**Reply by Trenkost** [reply]
I'm having this exact same issue except I have no trains on the network. When I add a train (through the blueprint, as the OP did) the requester station doesn't fire. This M variable is busting my balls and I'm trying to understand how to get it to fire appropriately.

**Reply by webhead1104** [reply]
Hey, did any of you figure out how to fix this?
